src/xheartbeat.cpp

```diff
@@ -6,24 +6,24 @@
 * The full license is in the file LICENSE, distributed with this software. *
 ****************************************************************************/
 
+#include <iterator>
+#include <string>
+
 #include "xheartbeat.hpp"
 #include "zmq_addon.hpp"
 #include "xmiddleware.hpp"
-#include <iterator>
 
 namespace xeus
 {
     xheartbeat::xheartbeat(zmq::context_t& context,
                            const std::string& transport,
                            const std::string& ip,
-                           const std::string& port)
+                           std::string& port)
```

Comment:
This parameter should remain constant. Add a public method that returns the socket port number. The implementation should use `sock.getsockopt( ZMQ_LAST_ENDPOINT, &port, &size );` where `port` is a char buffer. This method fills the buffer port with the complete endpoint, so a method for parsing it and extract the port number should be implemented in `xmiddleware`.